pkg/archive/diff.go

```diff
@@ -40,12 +42,30 @@ func UnpackLayer(dest string, layer ArchiveReader) (size int64, err error) {
 		// Normalize name, for safety and for a simple is-root check
 		hdr.Name = filepath.Clean(hdr.Name)
 
-		if !strings.HasSuffix(hdr.Name, "/") {
+		// Windows does not support filenames with colons in them. Ignore
+		// these files. This is not a problem though (although it might
+		// appear that it is). Let's suppose a client is running docker pull.
+		// The daemon it points to is Windows. Would it make sense for the
+		// client to be doing a docker pull Ubuntu for example (which has files
+		// with colons in the name under /usr/share/man/man3)? No, absolutely
+		// not as it would really only make sense that they were pulling a
+		// Windows image. However, for development, it is necessary to be able
+		// to pull Linux images which are in the repository.
```

Comment:
(Still in flux and under discussion), but the gist of the plan is.... Eventually the docker registry will have implicit knowledge about whether the image is a Linux or Windows-based image (or possibly common, but I think that's relatively unlikely). Once that is ready in the registry, the daemon will be in a position to know what it's pulling. However, today, I still need to test the Windows "pull" path, and the only thing I can pull is Linux based images. IOW, net result is once we have the registry solution, this will be a no-op and can be removed. For now, I'd really like to log it though.